www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml

```diff
@@ -1432,6 +1444,53 @@
       </rate>
     </limit_zone>
 
+    <errorpage type="ArrayField">
+      <name type="TextField">
+        <Required>Y</Required>
+      </name>
+      <statuscodes type="OptionField">
+        <multiple>Y</multiple>
+        <OptionValues>
+          <status400>400 Bad Request</status400>
+          <status401>401 Unauthorized</status401>
+          <status403>403 Forbidden</status403>
+          <status404>404 Not Found</status404>
+          <status405>405 Method Not Allowed</status405>
+	  <status407>407 Proxy Authentication Required</status407>
+	  <status408>408 Request Timeout</status408>
+	  <status410>410 Gone</status410>
+	  <status415>415 Unsupported Media Type</status415>
+          <status429>429 Too Many Requests</status429>
+          <status431>431 Request Header Fields Too Large</status431>
+          <status500>500 Internal Server Error</status500>
+          <status501>501 Not Implemented</status501>
+          <status502>502 Bad Gateway</status502>
+          <status503>503 Service Unavailable</status503>
+          <status504>504 Gateway Timeout</status504>
+        </OptionValues>
+        <Required>Y</Required>
+      </statuscodes>
+      <page type="TextField">
+        <Required>Y</Required>
```

Comment:
maybe limit the charset or generate and store that file in config.xml to have at least some validation?

Reply:
Added a mask of `/^[-./\w]+$/`.

In general I dislike having a file system dependency here as these error pages are not part of a config export, which somehow contradicts the . What's your opinion in adding the page content itself to the model? Is there a size limit of the model/complete config?

Reply:
I think the out of memory limit is the only valid limit. Since this files should not exceed some kilobytes, I think it is ok if they are stored in the config.xml as BASE64 content.

There is already some file content that is stored in the config.xml (I think it is the captive portal template).

Reply:
@fabianfrz Thanks for the hint with the CaptivePortal, which indeed contains a very extensive file upload mechanism. I think for now a simple self-containing HTML suffices. See f929358.
As a potential future improvement we can add ZIP upload (for HTML error pages with ext. resources).

Reply:
@8191 External files will not work as they do not run in the same transaction. The browser will fire the request against the standard backend and I try to avoid touching the traffic where possible. This is why I have encoded the files so I have standalone HTML files everywhere (containing all CSS, JS and Images needed).